File: src/java/org/elastos/hive/vault/scripting/AndCondition.java
package org.elastos.hive.vault.scripting;

import java.util.List;

/**
 * Vault script condition that succeeds only if all the contained conditions are successful.
 */
public class AndCondition extends AggregatedCondition {
	public AndCondition(String name, List<Condition> conditions) {
		super(name, Type.AND, conditions);
	}

	public AndCondition(String name) {
		this(name, null);
	}
}

File: src/service/scripting/andcondition.ts
import { AggregatedCondition } from "./aggregatedcondition";
import { Condition } from "./condition";

/**
 * Vault script condition that succeeds only if all the contained conditions are successful.
 */
export class AndCondition extends AggregatedCondition {
	public constructor(name: string, conditions?: Condition[]) {
		super(name, AggregatedCondition.AND, conditions);
	}
}
